packages/kbn-reporting/public/share/share_context_menu/register_csv_modal_reporting.tsx

```diff
@@ -0,0 +1,202 @@
+      absoluteTime,
+    }: {
+      addGlobalTimeFilter?: boolean;
+      absoluteTime?: boolean;
+    }) => SearchSourceFields;
+
+    const jobParams = {
+      title: sharingData.title as string,
+      objectType,
+    };
+
+    const getJobParams = (forShareUrl?: boolean) => {
+      if (reportType === CSV_JOB_TYPE_V2) {
+        // csv v2 uses locator params
+        return {
+          ...jobParams,
+          locatorParams: sharingData.locatorParams as [Record<string, unknown>],
+        };
+      }
+
+      // csv v1 uses search source and columns
+      return {
+        ...jobParams,
+        columns: sharingData.columns as string[] | undefined,
+        searchSource: getSearchSource({
+          addGlobalTimeFilter: true,
+          absoluteTime: !forShareUrl,
+        }),
+      };
+    };
+
+    const shareActions: ShareMenuItem[] = [];
+
+    const licenseCheck = checkLicense(license.check('reporting', 'basic'));
+    const licenseToolTipContent = licenseCheck.message;
+    const licenseHasCsvReporting = licenseCheck.showLinks;
+    const licenseDisabled = !licenseCheck.enableLinks;
+
+    let capabilityHasCsvReporting = false;
+    if (usesUiCapabilities) {
+      capabilityHasCsvReporting = application.capabilities.discover?.generateCsv === true;
+    } else {
+      capabilityHasCsvReporting = true; // deprecated
+    }
+
+    const generateReportingJobCSV = ({ intl: intlReport, toasts: toastsReport }: ShareContext) => {
+      const decoratedJobParams = apiClient.getDecoratedJobParams(getJobParams());
+      return apiClient
+        .createReportingJob(reportType, decoratedJobParams)
+        .then(() => {
+          toastsReport.addSuccess({
+            title: intlReport?.formatMessage(
+              {
+                id: 'reporting.share.modalContent.successfullyQueuedReportNotificationTitle',
+                defaultMessage: 'Queued report for {objectType}',
+              },
+              { objectType }
+            ),
+            text: toMountPoint(
+              <FormattedMessage
+                id="reporting.share.modalContent.successfullyQueuedReportNotificationDescription"
+                defaultMessage="Track its progress in {path}."
+                values={{
+                  path: (
+                    <a href={apiClient.getManagementLink()}>
+                      <FormattedMessage
+                        id="reporting.share.publicNotifier.reportLink.reportingSectionUrlLinkLabel"
+                        defaultMessage="Stack Management &gt; Reporting"
+                      />
+                    </a>
+                  ),
+                }}
+              />,
+              { theme$: theme.theme$ }
+            ),
+            'data-test-subj': 'queueReportSuccess',
+          });
+          if (onClose) {
+            onClose();
+          }
+        })
+        .catch((error) => {
+          toastsReport.addError(error, {
+            title: intlReport!.formatMessage({
```

Comment:
Use the `toasts` and `intl` passed to `getShareMenuItems`
```suggestion
          toasts.addError(error, {
            title: intl.formatMessage({
```